### PR TITLE
[Forge] Multi-region simulation test

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR adds support simulating varying bandwidth and latency between nodes in a `MultiRegionMultiCloudSimulation` forge test. While the test is general enough to simulate any number of regions, there are some limitations with the underlying dependency in ChaosMesh and/or Linux Traffic Control that prevents from simulating more than 4 regions currently.

As part of the change, this PR changes how bandwidth chaos is represented. It is now more similar to delay chaos in that it allows for simulation multiple bandwidth configurations between any source and target instances.

**Limitations**
Linux Traffic Control only supports a maximum of 16 qdisc (naively one can think of them as network channels) per interface, but three are reserved for general traffic tiering. Therefore, only 13 is available for simulation. Further, one can create a tree of qdisc inheriting parent qdisc that should theoretically allow simulating latency/bandwidth for at least 13 regions. However, it appears that some qdiscs cannot inherit others in certain situations thus leaving the support to only 4 regions for now.

To support more regions, the change also includes using a single configuration with direction `both` to simulate latency between two regions, instead of having two configurations with direction `from`.

### Test Plan

http://o11y.aptosdev.com/grafana/goto/-FgluSB4z?orgId=1

Repeat this for all network chaos resources:
`kubectl describe networkchaos/aws--sa-east-1-to-gcp--us-central1-delay -n <namespace>`
and make sure the status is `Selected` and `AllInjected`. 

### References
Some references for future work:
- https://tldp.org/HOWTO/Adv-Routing-HOWTO/lartc.qdisc.classful.html
- https://unix.stackexchange.com/questions/654559/can-i-use-multiple-qdiscs-per-device
- https://github.com/torvalds/linux/blob/2faac9a98f010cf5b342fa89ac489c4586364e6e/include/uapi/linux/pkt_sched.h#L145

### CODE REVIEW

File: testsuite/testcases/src/three_region_simulation_test.rs [hunk 1]
```diff
@@ -109,12 +86,19 @@ fn create_three_region_swarm_network_delay(swarm: &dyn Swarm) -> SwarmNetworkDel
...
+/// the same cloud with dedicated backbone like GCP
+fn create_bandwidth_limit(swarm: &dyn Swarm) -> SwarmNetworkBandwidth {
+    let all_validators = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
     SwarmNetworkBandwidth {
-        rate: 1000,
-        limit: 20971520,
-        buffer: 10000,
+        group_network_bandwidths: vec![GroupNetworkBandwidth {
+            name: "forge-namespace-1000mbps-bandwidth".to_owned(),
+            rate: 10000 / 8, // 1 Gbps in megabytes per second
```
Comment (rustielin) [inline testsuite/testcases/src/three_region_simulation_test.rs:96]: nit: update comment

File: testsuite/forge/src/backend/k8s/chaos/network_delay.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ spec:
   # to (the packets to target)
   # both ( the packets from or to target)
   # This parameter makes Chaos only take effect for a specific direction of packets.
```
Comment (rustielin) [inline testsuite/forge/src/backend/k8s/chaos/network_delay.yaml:22]: nit: edit this comment :p

**Review by rustielin — APPROVED** [review]
Amazing work! How are you confirming that the applied `NetworkChaos` is having the intended effect? I remember when originally applying some chaos, some combination of effects (and their mode/direction) have some confusing behavior

The test runner logs from Humio make sense though: [link here](https://cloud.us.humio.com/k8s/search?columns=%5B%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22%40timestamp%22%2C%22format%22%3A%22timestamp%22%2C%22width%22%3A180%7D%2C%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22message%22%2C%22format%22%3A%22text%22%2C%22width%22%3A1105%7D%5D&end=1679503909550&live=false&newestAtBottom=true&query=balaji-1669261449%20%7C%20k8s.container_name%20%3D%20%22main%22&showOnlyFirstLine=false&start=1679499049550&tz=America%2FLos_Angeles&widgetType=list-view), when checking the chaos k8s specs manually

**Review by danielxiangzl — APPROVED** [review]

File: testsuite/forge/src/backend/k8s/chaos/network_delay.yaml (hunk 2 above)
Comment (ibalajiarun) [reply testsuite/forge/src/backend/k8s/chaos/network_delay.yaml:22]: I believe this is generic line is generic enough. It only talks about possible values right?

**Comment by ibalajiarun** [comment]
> Amazing work! How are you confirming that the applied `NetworkChaos` is having the intended effect? I remember when originally applying some chaos, some combination of effects (and their mode/direction) have some confusing behavior
> 
> The test runner logs from Humio make sense though: [link here](https://cloud.us.humio.com/k8s/search?columns=%5B%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22%40timestamp%22%2C%22format%22%3A%22timestamp%22%2C%22width%22%3A180%7D%2C%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22message%22%2C%22format%22%3A%22text%22%2C%22width%22%3A1105%7D%5D&end=1679503909550&live=false&newestAtBottom=true&query=balaji-1669261449%20%7C%20k8s.container_name%20%3D%20%22main%22&showOnlyFirstLine=false&start=1679499049550&tz=America%2FLos_Angeles&widgetType=list-view), when checking the chaos k8s specs manually

@rustielin Updated the test plan in description. Essentially I checked the status of each chaos resource and made sure that it is injected. Also, if there are any errors, you can find them in the chaos daemon pods (which is what I used for debugging)

File: testsuite/forge/src/backend/k8s/chaos/network_delay.yaml (hunk 2 above)
Comment (rustielin) [reply testsuite/forge/src/backend/k8s/chaos/network_delay.yaml:22]: oh nevermind, ignore lol I'm misreading my own past comment

**Comment by rustielin** [comment]
@ibalajiarun gotcha, that makes sense! It certainly works for devs with context and the infra expertise to check the chaos daemon. I'm wondering if there's a more systematic way to check this from within the test setup itself, perhaps with some actual network calls

**Comment by ibalajiarun** [comment]
@rustielin good point. I've been exploring that. Let me do that as a follow up.

**Pull request merged**